tests/wallet/nft_wallet/test_nft_mint_from_did.py

```diff
@@ -0,0 +1,494 @@
+    await time_out_assert(10, wallet_taker.get_unconfirmed_balance, funds)
+    await time_out_assert(10, wallet_taker.get_confirmed_balance, funds)
+
+    api_maker = WalletRpcApi(wallet_node_maker)
+    api_taker = WalletRpcApi(wallet_node_taker)
+    config = bt.config
+    daemon_port = config["daemon_port"]
+
+    def stop_node_cb() -> None:
+        pass
+
+    full_node_rpc_api = FullNodeRpcApi(full_node_api.full_node)
+
+    rpc_cleanup_node, test_rpc_port_node = await start_rpc_server(
+        full_node_rpc_api,
+        self_hostname,
+        daemon_port,
+        uint16(0),
+        stop_node_cb,
+        bt.root_path,
+        config,
+        connect_to_daemon=False,
+    )
+
+    rpc_cleanup, test_rpc_port = await start_rpc_server(
+        api_maker,
+        self_hostname,
+        daemon_port,
+        uint16(0),
+        lambda x: None,  # type: ignore
+        bt.root_path,
+        config,
+        connect_to_daemon=False,
+    )
+    client = await WalletRpcClient.create(self_hostname, test_rpc_port, bt.root_path, config)
+    client_node = await FullNodeRpcClient.create(self_hostname, test_rpc_port_node, bt.root_path, config)
+
+    did_wallet_maker: DIDWallet = await DIDWallet.create_new_did_wallet(
+        wallet_node_maker.wallet_state_manager, wallet_maker, uint64(1)
+    )
+    spend_bundle_list = await wallet_node_maker.wallet_state_manager.tx_store.get_unconfirmed_for_wallet(
+        did_wallet_maker.id()
+    )
+
+    spend_bundle = spend_bundle_list[0].spend_bundle
+    await time_out_assert_not_none(5, full_node_api.full_node.mempool_manager.get_spendbundle, spend_bundle.name())
+
+    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph_token))
+
+    await time_out_assert(15, wallet_maker.get_pending_change_balance, 0)
+    await time_out_assert(10, wallet_maker.get_unconfirmed_balance, funds - 1)
+    await time_out_assert(10, wallet_maker.get_confirmed_balance, funds - 1)
+
+    hex_did_id = did_wallet_maker.get_my_DID()
+    hmr_did_id = encode_puzzle_hash(bytes32.from_hexstr(hex_did_id), AddressType.DID.hrp(config))
+
+    nft_wallet_maker = await api_maker.create_new_wallet(
+        dict(wallet_type="nft_wallet", name="NFT WALLET 1", did_id=hmr_did_id)
+    )
+    assert isinstance(nft_wallet_maker, dict)
+    assert nft_wallet_maker.get("success")
+
+    nft_wallet_taker = await api_taker.create_new_wallet(dict(wallet_type="nft_wallet", name="NFT WALLET 2"))
+
+    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph_token))
+
+    sample = {
+        "hash": bytes32(token_bytes(32)).hex(),
+        "uris": ["https://data.com/1234"],
+        "meta_hash": bytes32(token_bytes(32)).hex(),
+        "meta_uris": ["https://meatadata.com/1234"],
+        "license_hash": bytes32(token_bytes(32)).hex(),
+        "license_uris": ["https://license.com/1234"],
+        "series_numer": 1,
+        "series_total": 1,
```

Comment:
I believe these should be `edition_number` and `edition_total`? Would be nice if the test checked the contents of minted NFTs.

Reply:
Have updated test metadata to vary urls and edition numbers, and added a section in the test to confirm the NFT data and edition numbers match the input data.